frame/treasury/src/lib.rs

```diff
@@ -595,6 +713,190 @@ decl_module! {
+				ensure!(bounty.status == BountyStatus::Proposed, Error::<T>::UnexpectedStatus);
+
+				BountyDescriptions::remove(bounty_id);
+
+				let value = bounty.bond;
+				let imbalance = T::Currency::slash_reserved(&bounty.proposer, value).0;
+				T::ProposalRejection::on_unbalanced(imbalance);
+
+				Self::deposit_event(Event::<T>::BountyRejected(bounty_id, value));
+
+				*maybe_bounty = None;
+
+				Ok(())
+			})?;
+		}
+
+		/// Approve a bounty proposal. At a later time, the bounty will be funded and become active
+		/// and the original deposit will be returned.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - One DB change.
+		/// # </weight>
+		#[weight = 100_000_000]
+		fn approve_bounty(origin, #[compact] bounty_id: ProposalIndex) {
+			T::ApproveOrigin::ensure_origin(origin)?;
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+				ensure!(bounty.status == BountyStatus::Proposed, Error::<T>::UnexpectedStatus);
+
+				bounty.status = BountyStatus::Approved;
+
+				BountyApprovals::mutate(|v| v.push(bounty_id));
+
+				Ok(())
+			})?;
+		}
+
+		#[weight = 100_000_000]
+		fn award_bounty(origin, #[compact] bounty_id: ProposalIndex, beneficiary: <T::Lookup as StaticLookup>::Source) {
+			let curator = ensure_signed(origin)?;
+			let beneficiary = T::Lookup::lookup(beneficiary)?;
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+				ensure!(bounty.status.is_active(), Error::<T>::UnexpectedStatus);
+				ensure!(bounty.curator == curator, Error::<T>::RequireCurator);
+				bounty.status = BountyStatus::PendingPayout {
+					beneficiary: beneficiary.clone(),
+					unlock_at: system::Module::<T>::block_number() + T::BountyDepositPayoutDelay::get(),
+				};
+
+				Ok(())
+			})?;
+
+			Self::deposit_event(Event::<T>::BountyAwarded(bounty_id, beneficiary));
+		}
+
+		#[weight = 100_000_000]
+		fn claim_bounty(origin, #[compact] bounty_id: BountyIndex) {
+			let _ = ensure_signed(origin)?; // anyone can trigger claim
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let bounty = maybe_bounty.take().ok_or(Error::<T>::InvalidIndex)?;
+				if let BountyStatus::PendingPayout { beneficiary, unlock_at } = bounty.status {
+					ensure!(system::Module::<T>::block_number() >= unlock_at, Error::<T>::Premature);
+					let bounty_account = Self::bounty_account_id(bounty_id);
+					let balance = T::Currency::free_balance(&bounty_account);
+					let fee = bounty.fee;
+					let payout = balance.saturating_sub(fee);
+					let _ = T::Currency::transfer(&bounty_account, &bounty.curator, fee, AllowDeath); // should not fail
+					let _ = T::Currency::transfer(&bounty_account, &beneficiary, payout, AllowDeath); // should not fail
```

Comment:
Need to come back to this, but I dont like that we are creating a brand new account for a single bounty just to do this transfer. maybe it is explained later

Reply:
If this should not fail, I bet there is a better function to call than transfer. Seems like you want best effort movement of the funds.

Reply:
The reason we create a new account to hold fund for bounty is to allow people be able contribute into that bounty by simply transfer money into the bounty account. It will also allow to make treasury proposal to top up an existing bounty.

Reply:
There are few other places existing doing transfer like this, just search `let _ = T::Currency::transfer`